remote/bitbucketserver/internal/client.go

```diff
@@ -214,3 +250,142 @@ func (c *Client) paginatedRepos(start int) ([]*Repo, error) {
 	}
 	return repoResponse.Values, nil
 }
+
+func Filter(vs []string, f func(string) bool) []string {
+	vsf := make([]string, 0)
+	for _, v := range vs {
+		if f(v) {
+			vsf = append(vsf, v)
+		}
+	}
+	return vsf
+}
+
+//TODO: find a clean way of doing these next two methods- bitbucket server hooks only support 20 cb hooks
+func arrayToHookSettings(hooks []string) HookSettings {
```

Comment:
yea, I figured it had something to do with the stash api being a bit odd. I see nothing wrong with the current approach, since Go reflection might reduce the amount of code, but at the cost of readability.

another option could be just to unmarshal to a `map[string]string` so you can build the key dynamically with `map["hook-url-"+i]=...`